File: app/src/main/java/com/sound/keloomacaua/activities/ui/login/LoginActivity.java
package com.sound.keloomacaua.activities.ui.login;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.auth.UserProfileChangeRequest;
import com.sound.keloomacaua.R;
import com.sound.keloomacaua.activities.ui.game.CreateOrJoinActivity;

public class LoginActivity extends AppCompatActivity {

    Button registerButton;
    EditText usernameEditText;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        FirebaseAuth auth = FirebaseAuth.getInstance();

        registerButton = findViewById(R.id.register);
        usernameEditText = findViewById(R.id.username);

        registerButton.setOnClickListener(v -> {
            registerButton.setEnabled(false);
            //register
            String username = usernameEditText.getText().toString();
            auth.signInAnonymously().addOnCompleteListener(authResultTask -> {
                if (authResultTask.isSuccessful()) {
                    UserProfileChangeRequest profileChange = new UserProfileChangeRequest.Builder()
                            .setDisplayName(username)
                            .build();
                    FirebaseUser loggedInUser = authResultTask.getResult().getUser();
                    loggedInUser.updateProfile(profileChange).addOnCompleteListener(task -> {
                        if (task.isSuccessful()) {
                            startGameListActivity();
                        } else {
                            showLoginFailed();
                        }
                    });
                } else {
                    showLoginFailed();
                }
            });
        });

        //autologin if user is already signed in
        FirebaseUser user = auth.getCurrentUser();
        if (user != null) {
            startGameListActivity();
        }
    }

    private void showLoginFailed() {
        Toast.makeText(getApplicationContext(), R.string.login_failed, Toast.LENGTH_SHORT).show();
        registerButton.setEnabled(true);
    }

    private void startGameListActivity() {
        Intent intent = new Intent(getApplicationContext(), CreateOrJoinActivity.class);
        startActivity(intent);
        registerButton.setEnabled(true);
    }
}

File: app/src/main/java/com/sound/keloomacaua/activities/ui/login/LoginActivity.kt
package com.sound.keloomacaua.activities.ui.login

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.UserProfileChangeRequest
import com.sound.keloomacaua.R
import com.sound.keloomacaua.activities.ui.game.LobbyActivity

class LoginActivity : AppCompatActivity() {
    lateinit var registerButton: Button
    lateinit var usernameEditText: EditText
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        val auth = FirebaseAuth.getInstance()
        registerButton = findViewById(R.id.register)
        usernameEditText = findViewById(R.id.username)
        registerButton.setOnClickListener {
            registerButton.isEnabled = false
            //register
            val username = usernameEditText.text.toString()
            auth.signInAnonymously().addOnCompleteListener { authResultTask: Task<AuthResult> ->
                if (authResultTask.isSuccessful) {
                    val profileChange = UserProfileChangeRequest.Builder()
                        .setDisplayName(username)
                        .build()
                    val loggedInUser = authResultTask.result?.user
                        ?: throw IllegalStateException("authResult is successful but user is null.")
                    loggedInUser.updateProfile(profileChange)
                        .addOnCompleteListener { task: Task<Void?> ->
                            if (task.isSuccessful) {
                                startGameListActivity()
                            } else {
                                showLoginFailed()
                            }
                        }
                } else {
                    showLoginFailed()
                }
            }
        }

        //autologin if user is already signed in
        val user = auth.currentUser
        if (user != null) {
            startGameListActivity()
        }
    }

    private fun showLoginFailed() {
        Toast.makeText(applicationContext, R.string.login_failed, Toast.LENGTH_SHORT).show()
        registerButton.isEnabled = true
    }

    private fun startGameListActivity() {
        val intent = Intent(applicationContext, LobbyActivity::class.java)
        startActivity(intent)
        registerButton.isEnabled = true
    }
}